docs/naps/2-conda-based-packaging.md

```diff
@@ -0,0 +1,527 @@
+* Allow advanced users to install packages from PyPI or other local sources via drag-and-drop. This
+  presents many risks that could irreversibly disrupt the installation, though, so it needs to be
+  considered very carefully.
+
+### Milestone 4: Enabling in-app napari version updates
+
+Our first attempts [^in-app-update-pr] to implementing this consisted of a pop-up dialog that would
+notify the user about the available updates. Accepting the dialog would trigger the creation of a
+new environment (e.g. `napari-0.4.16`) alongside the existing one (e.g. `napari-0.4.15`). While it
+could be argued that this is not an "update" per se, but a new installation, it fulfills the same
+goal: the new version is available. If the user wants to get rid of the old one, an option would
+have been offered to auto-clean on success, or after not using it for some configurable time.
+
+However, having several versions of napari coexisting might risk questions like "which napari
+version is the default one?", "which one will be in charge of updating napari?", "what if the
+update logic changes over time?". There possible workarounds, but from the user perspective it could
+get confusing very fast.
+
+For that reason, we decided to slightly redesign our approach to in-app updates. We will still use
+several environments to keep things tidy and performant (see "Detailed description" for Milestone
+2 for more details), but the update checks will mimic something closer to a a server-client model:
+
+* The server: We will create a separate standalone package to handle updates outside of the
+  `napari` process. This will be designed in an application agnostic way while targeting the needs
+  of the napari project. It will allow the user to handle updates but also manage the existing
+  napari installations (investigate issues, remove old ones, clean expired caches, etc).
+* The client: each napari version will have a very basic notion about the "update server" and will
+  query for updates every time it starts, and also periodically while running. If one is found, it
+  the server will run the update on its own.
+
+The governance of the "server" part will still fall under the `napari` organization (much like
+`magicgui`) but it will be usable outside of the napari project, following the same philosophy we
+adopted for the constructor work.
+
+### Milestone 5: Deprecating Briefcase-based installers
+
+See the corresponding section in "Detailed description".
+
+## Backward Compatibility
+
+Migrating the backend on which we build the installers has little impact on the end users who
+just want to run napari. That said, `constructor` does not support the AppImage format for
+Linux or DMG for macOS, which were the ones previously used with Briefcase. We don't see this
+as a problem though, given the small number of downloads each format enjoyed in previous
+releases [^napari-releases-json].
```

Comment:
"enjoyed" :joy: :+1: 

Yeah dmg and pkg are both equally standard in macOS, I don't see this as a problem.

Reply:
dmg is download and unpack to use, delete directory to fully uninstall. pkg is old-style installer that could perform multiple actions on drive and you need to believe to pkg creator that it will put files in predicted places to allow uninstall.

Same problem like windows installers. 

This comment is only to notice that dmg and pkg are not equivalent.

Reply:
I haven't dug into the DMG/.app standards enough to know for sure, but my limited understanding (as @Czaki says) is that DMGs can only extract an .app bundle. This is tricky to support with `conda` because we need to know the path where it will be extracted to postprocess the files. We thought of doing this with a "first run only" startup script, but it felt very clunky. Supporting relocatability is tricky, but I think that .app's can be marked as non-relocatable. Anyway, we decided to go with PKGs because it's supported by constructor already and plays better with conda packaging. _If_ there's a very strong motivation to offer DMGs again, we _could_ think about it, but it's a non-trivial task.

Reply:
fwiw, I don't mind the pkg